ocs_ci/ocs/resources/pvc.py

```diff
@@ -184,6 +185,25 @@ def get_attached_pods(self):
                 attached_pods.append(pod_obj)
         return attached_pods
 
+    def create_pvc_snapshot(self, snap_yaml, snap_name, sc_name):
+        """
+        Create snapshot of a PVC
+        Args:
+            snap_yaml (str): The path of snapshot yaml
+            snap_name (str): The name of the snapshot to be created
+            sc_name (str): The name of the snapshot class
+        Returns:
+            OCS object
+        """
+        snapshot_data = templating.load_yaml(snap_yaml)
+        snapshot_data['metadata']['name'] = snap_name
+        snapshot_data['spec']['volumeSnapshotClassName'] = sc_name
+        snapshot_data['spec']['source']['persistentVolumeClaimName'] = self.name
+        ocs_obj = OCS(**snapshot_data)
```

Comment:
it is problematic since this is a class method of `PVC` which inherits `OCS`.

Better to have this as a standalone function and maybe even better would be to have a class representation for `Snapshot`, same as we have for `PVC`

Reply:
I thought of same but did like this because of a couple of reasons:

1. This is the property of PVC as we are taking the snapshot of PVC
2. Other than creating a class and creating an init function to create an OCS object there is nothing which is required from SNAP class so it would be like a dummy.
3. We are already calling OCS() in another PVC class property (backed_pv_obj)

Let me know if you still want the change, will create a separate class.

Reply:
Regarding #1 and #2, agree, we don't really need another class for snapshot then.
Regarding #3, we better fix this code then (backed_pv_obj). I suggest having the PVC creation function in this file but as a standalone function outside of PVC

Reply:
Fixed, PTAL